backend/migrations/versions/56ae3b6a8cf0_unique_application_name.py

```diff
@@ -0,0 +1,28 @@
+"""unique application name
+
+Revision ID: 56ae3b6a8cf0
+Revises: d7982c8b8229
+Create Date: 2022-11-02 17:11:08.177816
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '56ae3b6a8cf0'
+down_revision = 'd7982c8b8229'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic ###
+    op.create_unique_constraint(None, 'application', ['name'])
```

Comment:
Il ne faut pas donner un id à la contrainte ? J'ai l'impression que le downgrade ne va pas fonctionner si jamais il n'y a pas d'id défini mais peut-être qu'alembic est plus intelligent que ce que je pense.

Reply:
create unique constraint sur application.name semble être suffisant pour savoir de quoi on parle, pourquoi on aurait besoin d'un id ?
En tout cas, ça a été autogénéré comme ça, et la doc semble dire que l'autogénération marche pour ça  : 
Autogenerate will detect:

    Table additions, removals.

    Column additions, removals.

    Change of nullable status on columns.

    Basic changes in indexes and explicitly-named unique constraints

    Basic changes in foreign key constraints

Reply:
Ah je vois ce que tu veux dire, le drop constraint n'a pas le nom de la colonne..